R/tlf_ae_specific.R

```diff
@@ -67,57 +68,182 @@ tlf_ae_specific <- function(outdata,
       "Adverse event terms are from MedDRA Version {medra_version}."
     )
   }
+
   tbl <- outdata$tbl
   group <- outdata$group
   reference_group <- outdata$reference_group
+  display_total <- "Total" %in% group
   n_group <- length(outdata$group)
+  between_total <- seq(n_group - (display_total))
+  n_comparisons <- length(between_total) - 1
   n_row <- nrow(tbl)
   n_col <- ncol(tbl)
 
+  if (!is.null(col_rel_width) & !n_col == length(col_rel_width)) {
+    stop(
+      "col_rel_width must have the same length (has ",
+      length(col_rel_width),
+      ") as as outdata$tbl has number of columns (has ",
+      n_col, ")."
+    )
+  }
+
   # Define title
-  title <- collect_title(outdata$meta, outdata$population, outdata$observation, outdata$parameter, analysis = "ae_specific")
+  if (is.null(title)) {
+    title <- collect_title(outdata$meta,
+      outdata$population,
+      outdata$observation,
+      outdata$parameter,
+      analysis = "ae_specific"
+    )
+  }
+
   footnotes <- vapply(footnotes, glue::glue_data,
     .x = list(medra_version = medra_version), FUN.VALUE = character(1)
   )
   names(footnotes) <- NULL
 
-  # Define column header
-  colheader_n <- c(
-    paste0(" | ", paste(group, collapse = " | ")),
-    paste0(" | ", paste(rep("n | (%)", n_group), collapse = " | "))
+  # within (group statistics)
+
+  col_tbl_within <- strsplit(names(tbl), "_") |>
+    unlist() |>
+    (\(list) list[list %in% c("n", "prop", "dur", "events")])() |>
+    unique()
+
+  colhead_within <- paste(
+    sapply(
+      X = col_tbl_within,
+      FUN = switch,
+      "n" = "n",
+      "prop" = "(%)",
+      "dur" = "Mean Duration (SE)",
+      "events" = "Mean Events per Participant (SE)"
+    ),
+    collapse = " | "
+  )
+
+  colhead_1_within <- paste(group, collapse = " | ")
+
+  colhead_2_within <- paste(rep(colhead_within, n_group),
+    collapse = " | "
+  )
+
+  colborder_within <- sapply(
+    X = col_tbl_within,
+    FUN = switch,
+    "n" = "single",
+    "prop" = "",
+    "dur" = "single",
+    "events" = "single",
+    USE.NAMES = FALSE
+  )
+
+  rwidth_2_within <- rep(1, length(col_tbl_within) * n_group)
+
+  rwidth_1_within <- tapply(
+    rwidth_2_within,
+    c(rep(1:n_group, each = length(col_tbl_within))),
+    sum
   )
 
-  # TODO: add logic for CI and p-value with multipel groups following WMA mock up table.
-  # colheader_ci <- c(paste("Difference in % vs", group[reference_group]),
-  # "Estimate (95% CI)")
+  colborder_within <- rep(colborder_within, n_group)
 
-  # colheader_p <- c("", "p-value")
-  # colheader <- paste(colheader_n, colheader_ci, colheader_p, sep = " | ")
+  # between (comparison statistics)
 
-  colheader <- colheader_n
+  col_tbl_between <- strsplit(names(tbl), "_") |>
+    unlist() |>
+    (\(list) list[list %in% c("diff", "ci", "p")])() |>
+    unique()
+
+  if (length(col_tbl_between) > 0) {
```

Comment:
is there a nicer way to do this control?

Reply:
LGTM